src/librustc/lint/builtin.rs

```diff
@@ -252,7 +252,7 @@ declare_lint! {
 declare_lint! {
     pub ELIDED_LIFETIMES_IN_PATHS,
     Allow,
-    "hidden lifetime parameters are deprecated, try `Foo<'_>`"
+    "implicit lifetime parameters are deprecated"
```

Comment:
As far as names go, my take:

- `implicit` is *ok*, but it tends to mean anything that the reader wants it to mean
- `hidden` and `invisible` I think are both ok

I don't think any name will necessarily convey perfectly what we are going for -- but all 3 names are roughly in the right direction. I would summarize as "a lifetime that you can't tell is there from looking at the type". For this, I like "hidden" or "invisible" better than "implicit".